src/command/info/info.c

```diff
@@ -770,34 +1160,116 @@ infoRender(void)
         // Get stanza if specified
         const String *stanza = cfgOptionStrNull(cfgOptStanza);
 
+        // Initialize the list of stanzas on all repos
+        List *stanzaRepoList = lstNewP(sizeof(InfoStanzaRepo), .sortOrder = sortOrderAsc, .comparator = lstComparatorStr);
+
         // Get the backup label if specified
         const String *backupLabel = cfgOptionStrNull(cfgOptSet);
 
-        // Get the repo storage in case it is remote and encryption settings need to be pulled down
-        storageRepo();
-
-        // If a backup set was specified, see if the manifest exists
+        // Since the --set option depends on the --stanza option, the parser will error before this if the backup label is
+        // specified but a stanza is not
         if (backupLabel != NULL)
         {
             if (!strEq(cfgOptionStr(cfgOptOutput), CFGOPTVAL_INFO_OUTPUT_TEXT_STR))
-                THROW(ConfigError, "option 'set' is currently only valid for text output");
+                THROW(ConfigError, "option '" CFGOPT_SET "' is currently only valid for text output");
+
+            // !!! THIS LIMITATION SHOULD BE REMOVED
+            if (!(cfgOptionTest(cfgOptRepo)) && cfgOptionGroupIdxTotal(cfgOptGrpRepo) > 1)
+                THROW(OptionRequiredError, "option '" CFGOPT_REPO "' is required when specifying a backup set");
+        }
 
-            if (!storageExistsP(storageRepo(), strNewFmt(STORAGE_REPO_BACKUP "/%s/" BACKUP_MANIFEST_FILE, strZ(backupLabel))))
+        // Initialize the repo index
+        unsigned int repoIdxStart = 0;
+        unsigned int repoIdxMax = cfgOptionGroupIdxTotal(cfgOptGrpRepo);
+        unsigned int repoTotal = repoIdxMax;
+
+        // If the repo was specified then set index to the array location and max to loop only once
+        if (cfgOptionTest(cfgOptRepo))
+        {
+            repoIdxStart = cfgOptionGroupIdxDefault(cfgOptGrpRepo);
+            repoIdxMax = repoIdxStart + 1;
+        }
+
+        for (unsigned int repoIdx = repoIdxStart; repoIdx < repoIdxMax; repoIdx++)
+        {
+            // Get the repo storage in case it is remote and encryption settings need to be pulled down (performed here for testing)
```

Comment:
I'm not sure what "performed here for testing" means.

Reply:
fixed 8af20b6e1cacc2b4b68cbe5a4b7ab891ffa48768